cmdline.rs

```diff
@@ -0,0 +1,69 @@
+//! Command line-configurable transparent proxy.
+//!
+//! Useful for testing e.g. address translation or DNS related functionalities,
+//! such as alterations of hostname mapping combined with IP changes.
+//!
+//! How to use:
+//! cargo run --example cmdline -- [node_addr] [proxy_addr]
+//! This starts a proxy that forwards all traffic between a driver and [proxy_addr] to [node_addr].
+
+use std::{
+    io::Read as _,
+    net::{Ipv4Addr, SocketAddr},
+};
+
+use scylla_proxy::{Node, Proxy, ShardAwareness};
+use tracing::instrument::WithSubscriber;
+
+fn init_logger() {
+    tracing_subscriber::fmt::fmt()
+        .with_env_filter(tracing_subscriber::EnvFilter::from_default_env())
+        .without_time()
+        .init();
+}
+
+fn pause() {
+    println!("Press Enter to stop proxy...");
+    std::io::stdin().read_exact(&mut [0]).unwrap();
+    println!();
+}
```

Comment:
Ouch, you are using a blocking API here. The `pause()` invocation in `main` will block one of the executor threads.

How about waiting for a Ctrl+C signal instead? AFAIR tokio already has some utilities for that which allow to wait for this signal asynchronously.